system/jlib/jtrace.cpp

```diff
@@ -67,6 +71,143 @@ namespace opentel_trace = opentelemetry::trace;
+    void seed(SeedSequence &seed_sequence) noexcept
+    {
+        seed_sequence.generate(reinterpret_cast<uint32_t *>(state_.data()),
+                            reinterpret_cast<uint32_t *>(state_.data() + state_.size()));
+    }
+
+    static constexpr uint64_t(min)() noexcept { return 0; }
+
+    static constexpr uint64_t(max)() noexcept { return (std::numeric_limits<uint64_t>::max)(); }
+
+private:
+    std::array<uint64_t, 2> state_{};
+};
+/////////////////////////////from sdk/src/common/fast_random_number_generator.h//////////////////
+
+/////////////////////////////from sdk/src/common/random.cc///////////////////////////////////////
+class CleanTlsRandomNumberGenerator
+{
+public:
+    CleanTlsRandomNumberGenerator() noexcept
+    {
+        Seed();
+    }
+
+    NotSharedOTelFastRandomNumberGenerator &engine() noexcept { return engine_; }
+
+private:
+    NotSharedOTelFastRandomNumberGenerator engine_;
+
+    void Seed() noexcept
+    {
+        std::random_device random_device;
+        std::seed_seq seed_seq{random_device(), random_device(), random_device(), random_device()};
+        engine_.seed(seed_seq);
+    }
+};
+/////////////////////////////from sdk/src/common/random.cc///////////////////////////////////////
+
+/////////////////////////////from sdk/src/common/random.h///////////////////////////////////////
+class NotSharedOTelRandom
+{
+public:
+  /**
+   * @return an unsigned 64 bit random number
+   */
+    static uint64_t GenerateRandom64() noexcept
+    {
+        return GetRandomNumberGenerator()();
+    }
+
+    /**
+     * Fill the passed span with random bytes.
+     *
+     * @param buffer A span of bytes.
+     */
+    static void GenerateRandomBuffer(opentelemetry::nostd::span<uint8_t> buffer) noexcept
+    {
+        auto buf_size = buffer.size();
+
+        for (size_t i = 0; i < buf_size; i += sizeof(uint64_t))
+        {
+            uint64_t value = GenerateRandom64();
+            if (i + sizeof(uint64_t) <= buf_size)
+            {
+                memcpy(&buffer[i], &value, sizeof(uint64_t));
+            }
+            else
+            {
+                memcpy(&buffer[i], &value, buf_size - i);
+            }
+        }
+    }
+
+private:
+    /**
+     * @return a seeded thread-local random number generator.
+     */
+    static NotSharedOTelFastRandomNumberGenerator & GetRandomNumberGenerator() noexcept
+    {
+        static thread_local CleanTlsRandomNumberGenerator random_number_generator{};
+        return random_number_generator.engine();
+    }
+};
+/////////////////////////////from sdk/src/common/random.h///////////////////////////////////////
+
+class CustomIdGenerator : public opentelemetry::sdk::trace::IdGenerator
+{
+public:
+    CustomIdGenerator() : IdGenerator(true) {}
+
+    opentelemetry::trace::SpanId GenerateSpanId() noexcept override
+    {
+      uint8_t span_id_buf[trace_api::SpanId::kSize];
+      NotSharedOTelRandom::GenerateRandomBuffer(span_id_buf);
```

Comment:
Instead of mimicking OTel's default behavior, we're free to introduce any other random data generator here